static/js/apps/eval_retrieval_generation/sxs/app.tsx

```diff
@@ -24,23 +24,38 @@ import { getDocInfo } from "../util";
 import { SessionContext } from "./context";
 import { getLeftAndRight } from "./left_right_picker";
 import { QueryWithTables } from "./query_with_tables";
+import { SxsFeedback } from "./sxs_feedback";
 
 interface AppPropType {
   sessionId: string;
   sheetIdA: string;
   sheetIdB: string;
 }
 
+function getSortedQueryIds(docInfos: { a: DocInfo; b: DocInfo }) {
+  const idsA = Object.keys(docInfos?.a?.allQuery || []);
+  const idsB = Object.keys(docInfos?.b?.allQuery || []);
+  return idsA
+    .filter((id) => idsB.includes(id))
+    .map((id) => Number(id))
+    .sort((a, b) => a - b);
+}
+
 export function App(props: AppPropType): JSX.Element {
   const [user, setUser] = useState<User | null>(null);
   const [docInfos, setDocInfos] = useState<{ a: DocInfo; b: DocInfo }>(null);
   const { setSessionQueryId, sessionQueryId } = useContext(SessionContext);
+  const sortedQueryIds = getSortedQueryIds(docInfos);
+  if (!sessionQueryId) {
+    setSessionQueryId(sortedQueryIds[0]);
```

Comment:
check that sortedQueryIds is not empty first before accessing the first item?

Reply:
done